corehq/apps/builds/models.py

```diff
@@ -16,10 +16,13 @@ class SemanticVersionProperty(StringProperty):
 
     def validate(self, value, required=True):
         super(SemanticVersionProperty, self).validate(value, required)
+        if not self.required and not value:
+            return value
         try:
-            major, minor, _ = value.split('.')
+            major, minor, point = value.split('.')
             int(major)
             int(minor)
+            int(point)
```

Comment:
I had a though here, which is that this might interfere with legitimate version conventions for non-released versions (2.25.0a for alpha, 2.25.0b for beta, 2.25.0rc for release candidate), but if we're not using any of those right now, then I suppose we don't have to worry about it right now. Happy to make it strict now if current usage supports that, and then make it lax in the future if it's needed.